### PR TITLE
[BEAM-1871] Move several options/auth classes around in gcp-core

### PR DESCRIPTION
**Pull request by lukecwik**
Note that I kept a duplicate of GcsOptions because of its direct usage within the Dataflow Worker

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
Kicked off https://builds.apache.org/view/Beam/job/beam_PreCommit_Java_MavenInstall/10071/
R: @tgroh @kennknowles

**Comment by lukecwik** [comment]
Kicked off fix up build: https://builds.apache.org/view/Beam/job/beam_PreCommit_Java_MavenInstall/10073/

**Comment by lukecwik** [comment]
Failing tests seem to be from previous issues which implies that this change will be compatible with the Dataflow worker.

**Comment by lukecwik** [comment]
Rebase and rebuild: https://builds.apache.org/view/Beam/job/beam_PreCommit_Java_MavenInstall/10078/

**Comment by dhalperi** [comment]
LGTM and looks already green. Ship it?

File: sdks/java/extensions/gcp-core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java [hunk 1]
```diff
@@ -19,21 +19,17 @@
...
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
 import org.apache.beam.sdk.util.GcsPathValidator;
 import org.apache.beam.sdk.util.GcsUtil;
-import org.apache.beam.sdk.util.InstanceBuilder;
 import org.apache.beam.sdk.util.PathValidator;
 
 /**
  * Options used to configure Google Cloud Storage.
  */
+@Deprecated
```
Comment (tgroh) [inline sdks/java/extensions/gcp-core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: Note why deprecated in the comment

Comment (lukecwik) [inline sdks/java/extensions/gcp-core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: This is just a short lived object till I can update Dataflow Worker, plan to delete soon.

**Pull request closed without merging**